File: app/src/main/java/com/sortedqueue/programmercreek/adapter/ViewSlidesPagerAdapter.java
package com.sortedqueue.programmercreek.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.sortedqueue.programmercreek.database.SlideModel;
import com.sortedqueue.programmercreek.fragments.ViewSlideFragment;

import java.util.ArrayList;

/**
 * Created by Alok on 11/04/17.
 */

public class ViewSlidesPagerAdapter extends FragmentStatePagerAdapter {

    private ArrayList<SlideModel> slideModelArrayList;
    private ArrayList<ViewSlideFragment> fragmentArrayList;


    public ViewSlidesPagerAdapter(FragmentManager fm, ArrayList<SlideModel> slideModelArrayList) {
        super(fm);
        this.fragmentArrayList = new ArrayList<>();
        this.slideModelArrayList = slideModelArrayList;
        for( SlideModel slideModel : slideModelArrayList ) {
            ViewSlideFragment viewSlideFragment = new ViewSlideFragment();
            viewSlideFragment.setParameter( slideModel );
            fragmentArrayList.add(viewSlideFragment);
        }
    }

    @Override
    public Fragment getItem(int position) {
        return fragmentArrayList.get(position);
    }

    @Override
    public int getCount() {
        return slideModelArrayList.size();
    }
}

File: app/src/main/java/com/sortedqueue/programmercreek/adapter/ViewSlidesPagerAdapter.kt
package com.sortedqueue.programmercreek.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter

import com.sortedqueue.programmercreek.database.SlideModel
import com.sortedqueue.programmercreek.fragments.ViewSlideFragment

import java.util.ArrayList

/**
 * Created by Alok on 11/04/17.
 */

class ViewSlidesPagerAdapter(fm: FragmentManager, private val slideModelArrayList: ArrayList<SlideModel>) : FragmentStatePagerAdapter(fm) {
    private val fragmentArrayList: ArrayList<ViewSlideFragment>


    init {
        this.fragmentArrayList = ArrayList<ViewSlideFragment>()
        for (slideModel in slideModelArrayList) {
            val viewSlideFragment = ViewSlideFragment()
            viewSlideFragment.setParameter(slideModel)
            fragmentArrayList.add(viewSlideFragment)
        }
    }

    override fun getItem(position: Int): Fragment {
        return fragmentArrayList[position]
    }

    override fun getCount(): Int {
        return slideModelArrayList.size
    }
}
